base/client.jl

```diff
@@ -223,6 +235,11 @@ function exec_options(opts)
     global have_color     = (opts.color == 1) # --color=on
     global is_interactive = (opts.isinteractive != 0)
 
+    #check if stdout is TTY and color is not set manually
+    if !color_set && isa(stdout, TTY) && isa(stderr, TTY) && ttyhascolor()
```

Comment:
This is not the design described above. It said to set this the first time the variable is used, which is in stream.jl.